std/assembly/internal/memory.ts

```diff
@@ -186,67 +199,73 @@ export function memmove(dest: usize, src: usize, n: usize): void { // see: musl/
+    store<u8>(dest + n - 1, c);
+    if (n <= 2) return;
 
-  // fill head and tail with minimal branching
-  if (!n) return;
-  store<u8>(dest, c);
-  store<u8>(dest + n - 1, c);
-  if (n <= 2) return;
-
-  store<u8>(dest + 1, c);
-  store<u8>(dest + 2, c);
-  store<u8>(dest + n - 2, c);
-  store<u8>(dest + n - 3, c);
-  if (n <= 6) return;
-  store<u8>(dest + 3, c);
-  store<u8>(dest + n - 4, c);
-  if (n <= 8) return;
+    store<u8>(dest + 1, c);
+    store<u8>(dest + 2, c);
+    store<u8>(dest + n - 2, c);
+    store<u8>(dest + n - 3, c);
+    if (n <= 6) return;
+    store<u8>(dest + 3, c);
+    store<u8>(dest + n - 4, c);
+    if (n <= 8) return;
 
-  // advance pointer to align it at 4-byte boundary
-  var k: usize = -dest & 3;
-  dest += k;
-  n -= k;
-  n &= -4;
+    // advance pointer to align it at 4-byte boundary
+    let k: usize = -dest & 3;
+    dest += k;
+    n -= k;
+    n &= -4;
 
-  var c32: u32 = <u32>-1 / 255 * c;
+    let c32: u32 = <u32>-1 / 255 * c;
 
-  // fill head/tail up to 28 bytes each in preparation
-  store<u32>(dest, c32);
-  store<u32>(dest + n - 4, c32);
-  if (n <= 8) return;
-  store<u32>(dest + 4, c32);
-  store<u32>(dest + 8, c32);
-  store<u32>(dest + n - 12, c32);
-  store<u32>(dest + n - 8, c32);
-  if (n <= 24) return;
-  store<u32>(dest + 12, c32);
-  store<u32>(dest + 16, c32);
-  store<u32>(dest + 20, c32);
-  store<u32>(dest + 24, c32);
-  store<u32>(dest + n - 28, c32);
-  store<u32>(dest + n - 24, c32);
-  store<u32>(dest + n - 20, c32);
-  store<u32>(dest + n - 16, c32);
+    // fill head/tail up to 28 bytes each in preparation
+    store<u32>(dest, c32);
+    store<u32>(dest + n - 4, c32);
+    if (n <= 8) return;
+    store<u32>(dest + 4, c32);
+    store<u32>(dest + 8, c32);
+    store<u32>(dest + n - 12, c32);
+    store<u32>(dest + n - 8, c32);
+    if (n <= 24) return;
+    store<u32>(dest + 12, c32);
+    store<u32>(dest + 16, c32);
+    store<u32>(dest + 20, c32);
+    store<u32>(dest + 24, c32);
+    store<u32>(dest + n - 28, c32);
+    store<u32>(dest + n - 24, c32);
+    store<u32>(dest + n - 20, c32);
+    store<u32>(dest + n - 16, c32);
 
-  // align to a multiple of 8
-  k = 24 + (dest & 4);
-  dest += k;
-  n -= k;
+    // align to a multiple of 8
+    k = 24 + (dest & 4);
+    dest += k;
+    n -= k;
 
-  // copy 32 bytes each
-  var c64: u64 = <u64>c32 | (<u64>c32 << 32);
-  while (n >= 32) {
-    store<u64>(dest, c64);
-    store<u64>(dest + 8, c64);
-    store<u64>(dest + 16, c64);
-    store<u64>(dest + 24, c64);
-    n -= 32;
-    dest += 32;
+    // copy 32 bytes each
+    let c64: u64 = <u64>c32 | (<u64>c32 << 32);
+    while (n >= 32) {
+      store<u64>(dest, c64);
+      store<u64>(dest + 8, c64);
+      store<u64>(dest + 16, c64);
+      store<u64>(dest + 24, c64);
+      n -= 32;
+      dest += 32;
+    }
   }
 }
 
 export function memcmp(vl: usize, vr: usize, n: usize): i32 { // see: musl/src/string/memcmp.c
-  if (vl == vr) return 0;
+  if (vl === vr) return 0;
```

Comment:
nit: vl and vr aren't objects here so this isn't necessary

Reply:
I added `===` for consistency with `memmove` but I agree better use `==` instead. Fixed